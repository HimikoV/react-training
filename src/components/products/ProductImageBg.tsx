import { Grid, Box } from "@mui/material";
const ProductImageBg = () => {
  return (
    <Grid
      sx={{
        height: 315,
        width: "100%",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        overflow: "hidden",
        flexWrap: "nowrap",
      }}
    >
      <Box
        component="img"
        sx={{
          height: 1934,
          width: 315,
          transform: "rotate(270deg)",
          objectPosition: "48% 0",
          objectFit: "cover",
        }}
        alt="BG for products Page"
        src="https://images.unsplash.com/photo-1507546530-14a03f8d180a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=642&q=80"
      />
    </Grid>
  );
};

export default ProductImageBg;
