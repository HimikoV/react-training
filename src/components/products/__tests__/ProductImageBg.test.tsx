import ProductImageBg from "../ProductImageBg";

import renderer from "react-test-renderer";

describe("testing ProductImageBg render", () => {
  it("ProductImageBg Renders Correctly", () => {
    const productBg = renderer.create(<ProductImageBg />).toJSON();
    expect(productBg).toMatchSnapshot();
  });
});
