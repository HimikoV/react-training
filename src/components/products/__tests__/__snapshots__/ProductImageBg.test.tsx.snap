// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ProductImageBg render ProductImageBg Renders Correctly 1`] = `
<div
  className="MuiGrid-root css-b7ixwv-MuiGrid-root"
>
  <img
    alt="BG for products Page"
    className="MuiBox-root css-udfay2"
    src="https://images.unsplash.com/photo-1507546530-14a03f8d180a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=642&q=80"
  />
</div>
`;
